File: TeeNew/Builder2010/Marks_Positions.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Marks_Positions.h"
#include "TeeEditPro.hpp"
#include <TeeStore.hpp>
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeTools"
#pragma resource "*.dfm"
TMarksPositions *MarksPositions;
//---------------------------------------------------------------------------
__fastcall TMarksPositions::TMarksPositions(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TMarksPositions::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(5);
}
//---------------------------------------------------------------------------

void __fastcall TMarksPositions::Button1Click(TObject *Sender)
{
  SaveChartDialog(Chart1);
}
//---------------------------------------------------------------------------

void __fastcall TMarksPositions::Button2Click(TObject *Sender)
{
  if (OpenDialog1->Execute()); LoadChartFromFile(dynamic_cast<TCustomChart*>(Chart1),OpenDialog1->FileName);
}
//---------------------------------------------------------------------------

File: TeeNew/Marks_Positions.pas
unit Marks_Positions;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  QButtons,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  Buttons,
  {$ENDIF}
  Base, TeEngine, TeeTools, Series, TeeProcs, Chart, TeeGDIPlus;

type
  TMarksPositions = class(TBaseForm)
    Series1: TPieSeries;
    ChartTool1: TDragMarksTool;
    Button1: TSpeedButton;
    Button2: TSpeedButton;
    OpenDialog1: TOpenDialog;
    procedure FormCreate(Sender: TObject);
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeeEditPro, TeeStore;

procedure TMarksPositions.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(5);
end;

procedure TMarksPositions.Button1Click(Sender: TObject);
begin
  SaveChartDialog(Chart1);
end;

procedure TMarksPositions.Button2Click(Sender: TObject);
begin
  inherited;

  with OpenDialog1 do
       if Execute then
          LoadChartFromFile(TCustomChart(Chart1),FileName);
end;

initialization
  RegisterClass(TMarksPositions);
end.
